dcos/http/transport/roundtripper.go

```diff
@@ -172,6 +174,13 @@ func (t *dcosRoundtripper) CurrentToken() string {
 
 // RoundTrip is implementation of RoundTripper interface.
 func (t *dcosRoundtripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Set `User-Agent` header, defaulting to process name
+	userAgent := os.Args[0]
```

Comment:
I'm not sure if it's wise to expose the process name in HTTP requests by default. As a client of this library it would surprise me, and I worry (maybe I'm being paranoid) that `os.Args[0]` could be a filesystem path that contains something sensitive. These requests are probably only going to other DC/OS components, but user-agents tend to get logged. @dcos/security WDYT?

If we want to add a catch-all that helps us identify the DC/OS component that made a request, it might be more secure to use the name of the current process's systemd unit, and I think it'd be more useful in any case: https://godoc.org/github.com/coreos/go-systemd/util#CurrentUnitName It's probably a good idea to include the name of this library in the user-agent as well.

Here's what I think a useful default user-agent might look like for a request made from a process belonging to the `dcos-foo.service` systemd unit:
`User-Agent: dcos-go (systemd:dcos-foo.service)`

And for a request made in a process that isn't managed by systemd:
`User-Agent: dcos-go`

I'd also be OK with leaving the existing behavior of not setting a default user-agent.